src/reducers/order.js

```diff
@@ -0,0 +1,35 @@
+import { ADD_PRODUCT, REMOVE_PRODUCT, INCREASE_PRODUCT_QUANTITY, DECREASE_PRODUCT_QUANTITY, SAVE_ORDER } from "../types";
+
+const initialState = []
+
+export default function(state = initialState, action) {
+  switch (action.type) {
+    case ADD_PRODUCT:
+      return [
+          ...state,
+          action.payload
+      ]
+    case REMOVE_PRODUCT:
+      return state.filter((product, index) => {
+        return index !== action.payload.index
+      })
+    case INCREASE_PRODUCT_QUANTITY:
+      return state.map((product, index) => {
+        return index === action.payload.index ? { 
+          ...product,
+          quantity: product.quantity + 1 <= 10 ? product.quantity + 1 : 10
+         } : product
+      })
+    case DECREASE_PRODUCT_QUANTITY:
+      return state.map((product, index) => {
+        return index === action.payload.index ? { 
+          ...product,
+          quantity: product.quantity - 1 >= 1 ? product.quantity - 1 : 1
+         } : product
+      })
```

Comment:
This will get cleaned up if we decide to go with UPDATE_PRODUCT_QUANTITY, consider it